ipa-core/src/ff/boolean_array.rs

```diff
@@ -294,39 +301,29 @@ macro_rules! boolean_array_impl {
                 }
             }
 
-            // complains that no iter method exists, suppressed warnings
-            #[allow(clippy::into_iter_on_ref)]
             impl<'a> IntoIterator for &'a $name {
```

Comment:
I was wondering why there is no default implementation for IntoIterator if all that we do is just calling self.iter()?

Reply:
I think some tests still want it and I don't see harm adding it. Clippy is suggesting that calling `into_iter` on references may be confusing as it does not move the values out of the container. In our case, values are trivially copyable, so we can get away with it